lib/timers.js

```diff
@@ -276,7 +276,7 @@ exports.setInterval = function(callback, repeat) {
     timer._repeat();
 
     // Timer might be closed - no point in restarting it
-    if (!timer._repeat)
+    if ((!timer._repeat) || ((timer._idleTimeout === -1) && (!this._handle)))
```

Comment:
Checking `timer._idleTimeout === -1` is enough, if `_idleTimeout` should be set to `-1` in `Timeout.prototype.close` when `this._handle` is true.